synaptic_reconstruction/inference/cristae.py

```diff
@@ -0,0 +1,111 @@
+import time
+from typing import Dict, List, Optional, Tuple, Union
+import elf.parallel as parallel
+import numpy as np
+
+from skimage.transform import rescale, resize
+from util import get_prediction_torch_em
+
+DEFAULT_TILING = {
+    "tile": {"x": 512, "y": 512, "z": 64},
+    "halo": {"x": 64, "y": 64, "z": 8},
+}
+
+
+def _run_segmentation(
+    foreground, boundaries, verbose, min_size,
+    # blocking shapes for parallel computation
+    block_shape=(128, 256, 256),
+    halo=(48, 48, 48)
+):
```

Comment:
This function is too complex given that we only do foreground background segmentation. You don't need to run the watershed, it's enough to run connected components, on only the foreground predictions, do the size filter and then map the results back to a binary segmentation.